packages/base/src/errorwidget.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  WidgetModel,
+  DOMWidgetModel,
+  DOMWidgetView,
+  WidgetView
+} from './widget';
+import { JUPYTER_WIDGETS_VERSION } from './version';
+
+// create a Widget Model that captures an error object
+export function createErrorWidget(error: Error): typeof WidgetModel {
+  class ErrorWidget extends DOMWidgetModel {
+    constructor(attributes: any, options: any) {
+      attributes = {
+        ...attributes,
+        _view_name: 'ErrorWidgetView',
+        _view_module: '@jupyter-widgets/base',
+        _model_module_version: JUPYTER_WIDGETS_VERSION,
+        _view_module_version: JUPYTER_WIDGETS_VERSION,
+        failed_module: attributes._model_module,
+        failed_model_name: attributes._model_name,
+        error: error
+      };
+      super(attributes, options);
+      console.log(attributes);
+    }
+  }
+  return ErrorWidget;
+}
+
+export function createErrorWidgetView(error?: Error): typeof WidgetView {
```

Comment:
Do we actually need a WidgetView implementation? It's unlikely to be displayed on the page, or is it?

Reply:
It is, that's what you see in the screenshot. When the widget fails, you will see it on the page.